components/SnowOverlayToggle/SnowOverlayToggle.test.tsx

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+
+import { SnowOverlayToggle } from "./SnowOverlayToggle";
+import { render, screen } from '@testing-library/react';
+import { SnowOverlayProvider } from "@/providers/SnowOverlayProvider";
+import { userEvent } from "@storybook/test";
+
+
+describe('SnowOverlayToggle', () => {
+  it('Toggles the snowfall overlay', async ()=> {
+    render(
+      <SnowOverlayProvider>
+        <SnowOverlayToggle />
+      </SnowOverlayProvider>,
```

Comment:
refactor (blocking): the test is for the SnowOverlayToggle component, not the SnowOverlayToggle and SnowOverlayProvider.

Mock the SnowOverlayProvider in the tests and omit it from the render since we're focusing on unit tests at the moment for the tests for components. Having the provider is more of an integration test.

I'll defer to @shashilo though as I'm not sure of all the testing guidelines he provided to the team.

```suggestion
      <SnowOverlayToggle />
```